go-controller/pkg/ovn/ovn.go

```diff
@@ -124,18 +124,13 @@ func (oc *DefaultNetworkController) ensurePod(oldPod, pod *kapi.Pod, addPort boo
 		return nil
 	}
 
-	// skip the pods on no host subnet nodes
-	switchName := pod.Spec.NodeName
-	if oc.lsManager.IsNonHostSubnetSwitch(switchName) {
-		return nil
-	}
-
 	if oc.isPodScheduledinLocalZone(pod) {
 		klog.V(5).Infof("Ensuring zone local for Pod %s/%s in node %s", pod.Namespace, pod.Name, pod.Spec.NodeName)
 		return oc.ensureLocalZonePod(oldPod, pod, addPort)
 	}
 
-	klog.V(5).Infof("Ensuring zone remote for Pod %s/%s in node %s", pod.Namespace, pod.Name, pod.Spec.NodeName)
+	// Pods in no host subnet nodes can be seen as remote pods.
```

Comment:
ensureRemoteZonePod does more than just add ips to the namespace addr set.
Also, this will add pod IPs for all no hostsubnet nodes, not only during live migration. Not sure what are possible side-effects for that.

Reply:
Yes, ensureRemoteZonePod does more things. But it relies on some k8s specific pod annotations. In our case, those pod annotations don't exist for pods running on the non-ovn node. So I think we should be fine. But if you prefer, I can revise the code.